spec/operators/throttleTime-spec.ts

```diff
@@ -1,200 +1,254 @@
+        const t = time('   ----|                 ');
+        //                     ----|----|---|---|
+        const expected = '-a---y----b---x---x---(x|)';
+
+        const result = e1.pipe(throttleTime(t, rxTest, { leading: true, trailing: true }));
+
+        expectObservable(result).toBe(expected);
+        expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      });
     });
 
     it('should emit the value if only a single one is given', () => {
-      const e1 =   hot('-a--------------------|');
-      const t =   time('----|                  ');
-      const expected = '-a--------------------|';
+      rxTest.run(({ hot, time, expectObservable }) => {
+        const e1 = hot('  -a--------------------|');
+        const t = time('   ----|                 ');
+        const expected = '-a--------------------|';
 
-      const result = e1.pipe(throttleTime(t, rxTestScheduler, { leading: true, trailing: true }));
+        const result = e1.pipe(throttleTime(t, rxTest, { leading: true, trailing: true }));
 
-      expectObservable(result).toBe(expected);
+        expectObservable(result).toBe(expected);
+      });
     });
   });
 
   describe('throttleTime(fn, { leading: false, trailing: true })', () => {
-    it('should immediately emit the last value in each time window', () =>  {
-      const e1 =   hot('-a-xy-----b--x--cxxx--|');
-      const e1subs =   '^                     !';
-      const t =  time( '----|                  ');
-      const expected = '-----y--------x-----x-|';
-
-      const result = e1.pipe(throttleTime(t, rxTestScheduler, { leading: false, trailing: true }));
-
-      expectObservable(result).toBe(expected);
-      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    it('should immediately emit the last value in each time window', () => {
+      rxTest.run(({ hot, time, expectObservable, expectSubscriptions }) => {
+        const e1 = hot('  -a-xy-----b--x--cxxx--|');
+        const e1subs = '  ^---------------------!';
+        const t = time('   ----|                 ');
+        //                 ----|---|----|---|---|
+        const expected = '-----y--------x---x---(x|)';
+
+        const result = e1.pipe(throttleTime(t, rxTest, { leading: false, trailing: true }));
+
+        expectObservable(result).toBe(expected);
+        expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      });
     });
 
     it('should emit the last throttled value when complete', () => {
-      const e1 =   hot('-a-xy-----b--x--cxx|');
-      const e1subs =   '^                  !';
-      const t =   time('----|               ');
-      const expected = '-----y--------x----(x|)';
+      rxTest.run(({ hot, time, expectObservable, expectSubscriptions }) => {
+        const e1 = hot('  -a-xy-----b--x--cxx|');
+        const e1subs = '  ^------------------!';
+        const t = time('   ----|              ');
+        //                 ----|---|----|---|---|
+        const expected = '-----y--------x---x---|';
```

Comment:
question/nitpick: Should this really delay/throttle the completion of the source? This isn't the behaviour that I would have expected. Whatever the case, this test's description doesn't seem to describe what it's actually testing, IMO. If this is how it's supposed to behave, I think the description should explicitly state that complete notifications that are received whilst throttling will be delayed.

Reply:
Ah... yeah, I think if it has to trailing value we should probably complete it.. I'll make that change and merge.